rotkehlchen/tests/unit/globaldb/test_upgrades.py

```diff
@@ -382,6 +394,50 @@ def test_upgrade_v4_v5(globaldb):
         assert 'name' not in columns, 'The name column should not be in the contract_data table'
 
 
+@pytest.mark.parametrize('globaldb_upgrades', [[]])
+@pytest.mark.parametrize('reload_user_assets', [False])
+def test_upgrade_v5_v6(globaldb):
+    """Test the global DB upgrade from v5 to v6"""
+    # Check the state before upgrading
+    assert globaldb.get_setting_value('version', None) == 5
+    with globaldb.conn.read_ctx() as cursor:
+        # check existence of unique_cache table
+        cursor.execute(
+            'SELECT COUNT(*) FROM sqlite_master WHERE type="table" and name=?',
+            (UNIQUE_CACHE_TABLE_NAME,),
+        )
+        assert cursor.fetchone()[0] == 0
+
+    # execute upgrade
+    with ExitStack() as stack:
+        patch_for_globaldb_upgrade_to(stack, 6)
+        maybe_upgrade_globaldb(
+            connection=globaldb.conn,
+            global_dir=globaldb._data_directory / 'global_data',
+            db_filename=GLOBAL_DB_FILENAME,
+        )
+    assert globaldb.get_setting_value('version', None) == 6
+    with globaldb.conn.read_ctx() as cursor:
+        # check again existence of unique_cache table
+        cursor.execute(
+            'SELECT COUNT(*) FROM sqlite_master WHERE type="table" and name=?',
+            (UNIQUE_CACHE_TABLE_NAME,),
+        )
+        assert cursor.fetchone()[0] == 1
+        # check that appropriate cache data is transfered
+        for key_part in UNIQUE_CACHE_KEYS:
```

Comment:
We should better check the exact values. So:
1. Read the values from general_cache before the upgrade
2. Check that the cache is not empty and has some values for both general cache and unique cache
3. Upgrade
4. Based on the read data in step (1) compute the expected data and compare it to the entries in general_cache in unique_cache tables

This way we will be totally sure that:
1. Unique cache data was moved from general cache to unique cache
2. Data that had to stay in general_cache indeed stayed there and was not lost

Reply:
I followed your steps exactly